packages/http/src/mocker/negotiator/NegotiatorHelpers.ts

```diff
@@ -0,0 +1,266 @@
+    if (lowest2xxResponse) {
+      return helpers.negotiateOptionsBySpecificResponse(
+        httpOperation,
+        desiredOptions,
+        lowest2xxResponse
+      );
+    }
+    throw new Error('No 2** response defined, cannot mock');
+  },
+
+  negotiateOptionsBySpecificCode(
+    httpOperation: IHttpOperation,
+    desiredOptions: {
+      readonly mediaType?: string;
+      readonly code?: string;
+      readonly exampleKey?: string;
+      readonly dynamic?: boolean;
+    },
+    code: string
+  ): IHttpNegotiationResult {
+    // find response by provided status code
+    const responseByForcedStatusCode = findResponseByStatusCode(httpOperation.responses, code);
+    if (responseByForcedStatusCode) {
+      try {
+        // try to negotiate
+        return helpers.negotiateOptionsBySpecificResponse(
+          httpOperation,
+          desiredOptions,
+          responseByForcedStatusCode
+        );
+      } catch (error) {
+        // if negotiations fail try a default code
+        return helpers.negotiateOptionsForDefaultCode(httpOperation, desiredOptions);
+      }
+    }
+    // if no response found under a status code, try to mock a default code
+    return helpers.negotiateOptionsForDefaultCode(httpOperation, desiredOptions);
+  },
+
+  negotiateOptionsForValidRequest(
+    httpOperation: IHttpOperation,
+    desiredOptions: {
+      readonly mediaType?: string;
+      readonly code?: string;
+      readonly exampleKey?: string;
+      readonly dynamic?: boolean;
+    }
+  ): IHttpNegotiationResult {
+    const { code } = desiredOptions;
+    if (code) {
+      return helpers.negotiateOptionsBySpecificCode(httpOperation, desiredOptions, code);
+    }
+    return helpers.negotiateOptionsForDefaultCode(httpOperation, desiredOptions);
+  },
+
+  negotiateOptionsForInvalidRequest(httpResponses: IHttpResponse[]): IHttpNegotiationResult {
+    // currently only try to find a 400 response, but we may want to support other cases in the future
+    const code = '400';
+    const response = findResponseByStatusCode(httpResponses, code);
+    // TODO: what if no 400 response is defined?
+    if (!response) {
+      throw new Error('No 400 response defined');
+    }
+    // find first response with any static examples
+    const responseWithExamples = response.content.find(
+      content => !!content.examples && content.examples.length !== 0
+    );
+    // find first response with a schema
+    const responseWithSchema = response.content.find(content => !!content.schema);
+
+    if (responseWithExamples) {
+      return {
+        code,
+        mediaType: responseWithExamples.mediaType,
+        example: responseWithExamples.examples![0],
+      };
+    } else if (responseWithSchema) {
+      return {
+        code,
+        mediaType: responseWithSchema.mediaType,
+        schema: responseWithSchema.schema,
+      };
+    } else {
+      throw new Error('Data corrupted');
+    }
+  },
+};
+
+export default helpers;
```

Comment:
Can we move negotiator package outside of mocker again? It is a pretty cool package and we can use it to resolve other config options like if validation should be ran or not, or if the router should not match resources.servers hosts.